src/main/java/com/google/sps/data/ComputeAPIClient.java

```diff
@@ -0,0 +1,243 @@
+  public Operation createInstance(String instanceName, String sessionId)
+    throws IOException {
+      Instance instance = new Instance();
+      instance.setName(instanceName);
+      instance.setMachineType(String.format(
+        "projects/taniecer-step-2020/zones/us-central1-f/machineTypes/n1-standard-1",
+        PROJECT_ID, ZONE_NAME));
+      NetworkInterface ifc = new NetworkInterface();
+      ifc.setNetwork(String.format(
+        "https://www.googleapis.com/compute/v1/projects/%s/global/networks/default",
+        PROJECT_ID));
+      List<AccessConfig> configs = new ArrayList<>();
+      AccessConfig config = new AccessConfig();
+      config.setType(NETWORK_INTERFACE_CONFIG);
+      config.setName(NETWORK_ACCESS_CONFIG);
+      configs.add(config);
+      ifc.setAccessConfigs(configs);
+      instance.setNetworkInterfaces(Collections.singletonList(ifc));
+      AttachedDisk disk = new AttachedDisk();
+      disk.setBoot(true);
+      disk.setAutoDelete(true);
+      disk.setType("PERSISTENT");
+      AttachedDiskInitializeParams params = new AttachedDiskInitializeParams();
+      params.setDiskName(instanceName);
+      params.setSourceImage(SOURCE_IMAGE_PREFIX + SOURCE_IMAGE_PATH);
+      params.setDiskType(String.format(
+        "projects/taniecer-step-2020/zones/us-central1-a/diskTypes/pd-standard",
+        PROJECT_ID, ZONE_NAME));
+      disk.setInitializeParams(params);
+      instance.setDisks(Collections.singletonList(disk));
+      // Adds startup script to be used by VM Instance.
+      Metadata meta = new Metadata();
+      Metadata.Items item = new Metadata.Items();
+      item.setKey("startup-script-url");
+      item.setValue(String.format("gs://%s/starter_script.sh", PROJECT_ID));
+      meta.setItems(Collections.singletonList(item));
+      Metadata.Items item2 = new Metadata.Items();
+      item2.setKey("session-id");
+      item2.setValue(sessionId);
+      meta.setItems(Collections.singletonList(item2));
+      instance.setMetadata(meta);
+      ServiceAccount account = new ServiceAccount();
+      account.setEmail("default");
+      List<String> scopes = new ArrayList<>();
+      scopes.add("https://www.googleapis.com/auth/devstorage.full_control");
+      scopes.add("https://www.googleapis.com/auth/compute");
+      account.setScopes(scopes);
+      instance.setServiceAccounts(Collections.singletonList(account));
+      System.out.println(instance.toPrettyString());
+      Compute.Instances.Insert create = 
+        compute.instances().insert(PROJECT_ID, ZONE_NAME, instance);
+      return create.execute();
+    } 
+
+  public Operation stopInstance(String instanceName) throws Exception {
+    Compute.Instances.Stop stop =
+      compute.instances().stop(PROJECT_ID, ZONE_NAME, instanceName);
+    return stop.execute();
+  }
+
+  public Operation restartInstance(String instanceName, String sessionId) 
+    throws Exception {
+      // Adds startup script to be used by VM Instance.
```

Comment:
I think you can just get the instance and just reset its metadata
see: https://developers.google.com/resources/api-libraries/documentation/compute/v1/java/latest/com/google/api/services/compute/Compute.Instances.Get.html#Get-java.lang.String-java.lang.String-java.lang.String-

Reply:
i think what you have is starting the process of creating a new instance?